AirCasting/SessionViews/SessionHeaderView.swift

```diff
@@ -38,8 +38,11 @@ struct SessionHeaderView: View {
                     DeleteView(viewModel: DefaultDeleteSessionViewModel(session: session, measurementStreamStorage: measurementStreamStorage, streamRemover: DefaultSessionUpdateService(authorization: authorization, urlProvider: urlProvider), sessionSynchronizer: sessionSynchronizer), deleteModal: $showDeleteModal)
                 }
                 .sheet(isPresented: $showShareModal) {
-                    ShareSessionView(viewModel: DefaultShareSessionViewModel(session: session, exitRoute: {
+                    ShareSessionView(viewModel: DefaultShareSessionViewModel(session: session, apiClient: ShareSessionApi(urlProvider: urlProvider), exitRoute: { sharedEmail in
                         showShareModal.toggle()
+                        if sharedEmail == true {
+                            alert = InAppAlerts.shareFileRequestSent()
+                        }
```

Comment:
I need to say that the alert is not showing on my iPhone with iOS 15.2. I made some changes, and it is working now. I will paste the code here if you would like to use it :). I should mentioned also that on the Simulator, the alert works as expected.

```
 @State var detectEmailSent = false
    
    var body: some View {
        if #available(iOS 15, *) {
            sessionHeader
                .sheet(isPresented: $showDeleteModal) {
                    DeleteView(viewModel: DefaultDeleteSessionViewModel(session: session, measurementStreamStorage: measurementStreamStorage, streamRemover: DefaultSessionUpdateService(authorization: authorization, urlProvider: urlProvider), sessionSynchronizer: sessionSynchronizer), deleteModal: $showDeleteModal)
                }
                .sheet(isPresented: $showShareModal) {
                    ShareSessionView(viewModel: DefaultShareSessionViewModel(session: session, apiClient: ShareSessionApi(urlProvider: urlProvider), exitRoute: { sharedEmail in
                        showShareModal.toggle()
                        if sharedEmail == true {
                            detectEmailSent = true
                        }
                    })).onDisappear(perform: {
                        if detectEmailSent {
                            alert = InAppAlerts.shareFileRequestSent()
                        }
                    })
                }
```

Reply:
So with the current version, you don't see the alert on your iPhone, but you see it on the simulator? Or do you see it in the simulator after your changes?

Reply:
After the changes I can see it on my iPhone. Without changing anything, only using the simulator I can see the alert